lewis/core/statemachine.py

```diff
@@ -294,6 +295,8 @@ def bind_handlers_by_name(self, instance, override=False, prefix=None):
         """
         if prefix is None:
             prefix = {}
+        if not isinstance(prefix, dict) and hasattr(prefix, '__iter__'):
+            prefix = dict(zip(['on_entry', 'in_state', 'on_exit'], prefix))
 
         # Merge prefix defaults with any provided prefixes
         prefix = dict(list(self._prefix.items()) + list(prefix.items()))
```

Comment:
Hm. In principle yes, but I'm a bit concerned that this isn't currently `self._prefix.update(prefix)`... is it really functionally equivalent?